### PR TITLE
Added MITRE technique name to the alerts

### PR DESCRIPTION
**Pull request by danimegar**
|Related issue|
|---|
|#5203|

## Description

I added MITRE technique's name to the alerts. For that, I added a column in the attack table in mitre.db. Then I created a end-point in the Wazuh-DB to query a technique's name. This name is included in a Hash table with the MITRE tactics. Finally, the names and tactics are included in MITRE JSON object after consulting the Hash table.

## Configuration options

```
  <rule id="5712" level="10" frequency="8" timeframe="120" ignore="60">
    <if_matched_sid>5710</if_matched_sid>
    <description>sshd: brute force trying to get access to </description>
    <description>the system.</description>
    <mitre>
      <id>T1110</id>
    </mitre>
    <same_source_ip />
    <group>authentication_failures,pci_dss_11.4,pci_dss_10.2.4,pci_dss_10.2.5,gdpr_IV_35.7.d,gdpr_IV_32.2,hipaa_164.312.b,nist_800_53_SI.4,nist_800_53_AU.14,nist_800_53_AC.7,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
  </rule>
```

## Logs/Alerts example

```
{
   "timestamp":"2020-06-10T18:27:22.815+0200",
   "rule":{
      "level":10,
      "description":"sshd: brute force trying to get access to the system.",
      "id":"5712",
      "mitre":{
         "id":[
            "T1110",
            "T1021"
         ],
         "tactic":[
            "Credential Access",
            "Lateral Movement"
         ],
         "technique":[
            "Brute Force",
            "Remote Services"
         ]
      },
...
}
```

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux

- [x] Source installation
- [x] Package installation

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)

- [x] The data flow works as expected (agent-manager-api)
- [x] Added unit tests (for new features)
- [x] Integration tests

### CODE REVIEW

**Comment by danimegar** [comment]
## Unit tests

```
cd wazuh/src/unit_tests/build/analysisd
# ./test_mitre
```
image

All tests passed

**Comment by danimegar** [comment]
## Integration tests

```
cd wazuh-qa/tests/integration/
python3 -m pytest -v test_integration/test_analysisd/test_mitre/test_mitre_check_alert.py 
```
image

All tests passed

**Comment by danimegar** [comment]
## Source installation
### DEB

```
Synchronizing state of wazuh-manager.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable wazuh-manager
Starting Wazuh...

 - Configuration finished properly.

 - To start Wazuh:
      /var/ossec/bin/ossec-control start

 - To stop Wazuh:
      /var/ossec/bin/ossec-control stop

 - The configuration can be viewed or modified at /var/ossec/etc/ossec.conf
```

### RPM

```
Starting Wazuh...

 - Configuration finished properly.

 - To start Wazuh:
      /var/ossec/bin/ossec-control start

 - To stop Wazuh:
      /var/ossec/bin/ossec-control stop

 - The configuration can be viewed or modified at /var/ossec/etc/ossec.conf
```

Source installations correctly worked.

**Comment by danimegar** [comment]
## Package installation
### DEB

### RPM

**Comment by danimegar** [comment]
## Scan-build
image

One error found but it is not related to the issue:

image

**Comment by danimegar** [comment]
## Valgrind
Valgrind was run using different combination such as:
```
    <mitre>
      <id>T1098</id>
    </mitre>
```
```
    <mitre>
      <id>T1098</id>
      <id>T1015</id>
    </mitre>
```
```
    <mitre>
      <id>T1098</id>
      <id>T1015</id>
      <id>T1015</id>
    </mitre>
```
No memory leaks found

```
==17057== LEAK SUMMARY:
==17057==    definitely lost: 100 bytes in 9 blocks
==17057==    indirectly lost: 0 bytes in 0 blocks
==17057==      possibly lost: 0 bytes in 0 blocks
==17057==    still reachable: 13,255 bytes in 312 blocks
==17057==         suppressed: 0 bytes in 0 blocks
```

**Comment by danimegar** [comment]
## Coverity

```
Build ID: 320942

    Analysis Summary:
       New defects found: 7
       Defects eliminated: 0
```
The 7 new defects are type `Dereference null return value` and they are related to this issue.

**Comment by danimegar** [comment]
## Executing tests related to MITRE from wazuh-api

```
[root@centos7-manager1 wazuh-api]# mocha test/test_mitre.js 
(node:31551) [DEP0006] DeprecationWarning: child_process: options.customFds option is deprecated. Use options.stdio instead.

  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․

  32 passing (12s)
```

File: src/analysisd/mitre.c [hunk 1]
```diff
@@ -110,15 +111,34 @@ int mitre_load(char * mode){
...
         tactics_json = NULL;
 
+        /* Consulting Mitre's database to get technique's name */
+        snprintf(wazuhdb_query, OS_SIZE_6144, "mitre get name %s", ext_id);
+        if (result = wdbc_query_ex(&sock, wazuhdb_query, response, OS_SIZE_6144), result == -2) {
+            merror("Unable to connect to socket '%s'", WDB_LOCAL_SOCK);
+            goto end;
+        }
+
+        if (result == -1) {
```
Comment (vikman90) [inline src/analysisd/mitre.c:121]: We should call `wdbc_parse_result` before handling the buffer `response`.

**Pull request merged**